<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>There are several reasons to use a group in a regular expression:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>to change the precedence (e.g. <code>do(g|or)</code> will match 'dog' and 'door')</p>
</li>
<li>
<p>to remember parenthesised part of the match in the case of capturing group</p>
</li>
<li>
<p>to improve readability</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>In any case, having an empty group is most probably a mistake. Either it is a leftover after refactoring and should be removed, or the actual parentheses were intended and were not escaped.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">"foo()"  // Noncompliant, will match only 'foo'</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-java" data-lang="java">"foo\\(\\)"  // Matches 'foo()'</code></pre>
</div>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_message">Message</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Remove this empty group.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_highlighting">Highlighting</h2>
<div class="sectionbody">
<div class="paragraph">
<p>The empty group should be highlighted.</p>
</div>
<hr>
</div>
</div>