<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>To avoid holding more connections than necessary and to avoid potentially exhausting the number of available sockets when using <code>HttpClient</code>, <code>DocumentClient</code>, <code>QueueClient</code>, <code>ConnectionMultiplexer</code> or Azure Storage clients, consider:</p>
</div>
<div class="ulist">
<ul>
<li>
<p>Creating a single, thread-safe static client that every Azure Function invocation can use. Provide it in a shared class when different Azure Functions need it.</p>
</li>
<li>
<p>Instantiate the client as a thread-safe Singleton or a pool of reusable instances and use it with dependency injection.</p>
</li>
</ul>
</div>
<div class="paragraph">
<p>These classes typically manage their own connections to the resource, and thus are intended to be instantiated once and reused throughout the lifetime of an application.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">    public class HttpExample
    {
        [FunctionName("HttpExample")]
        public async Task&lt;IActionResult&gt; Run([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest request)
        {
            HttpClient httpClient = new HttpClient(); // Noncompliant

            var response = await httpClient.GetAsync("https://example.com");
            // rest of the function
        }
    }</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-csharp" data-lang="csharp">    public class HttpExample
    {
        [FunctionName("HttpExample")]
        public async Task&lt;IActionResult&gt; Run([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest request, IHttpClientFactory clientFactory)
        {
            var httpClient = clientFactory.CreateClient();
            var response = await httpClient.GetAsync("https://example.com");
            // rest of the function
        }
    }</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_see">See</h2>
<div class="sectionbody">
<div class="ulist">
<ul>
<li>
<p><a href="https://docs.microsoft.com/en-us/azure/azure-functions/manage-connections?tabs=csharp#static-clients">Manage connections in Azure Functions: Static Clients</a></p>
</li>
<li>
<p><a href="https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-dependency-injection#service-lifetimes">Azure Functions - Dependency Injection: Service Lifetimes</a></p>
</li>
<li>
<p><a href="https://docs.microsoft.com/en-us/azure/architecture/antipatterns/improper-instantiation/">Improper Instantiation antipattern</a></p>
</li>
</ul>
</div>
</div>
</div>