<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>When allocating memory with <code>malloc</code>, <code>calloc</code> and <code>realloc</code> it is important to make sure that the size of the allocated memory is compatible with the receiver type.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f() {
  ...
  int* ptr1 = malloc(sizeof(short)); // Noncompliant
  ...
}</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cpp" data-lang="cpp">void f() {
  ...
  int* ptr1 = malloc(sizeof(int));
  ...
}</code></pre>
</div>
</div>
</div>
</div>