<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Condition operators can quickly become hard to be read.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">function sayHello() {
  var a = (a === 'A') ? 'is A' : 'is not A'; // Noncompliant
}</code></pre>
</div>
</div>
</div>
</div>