<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Executing a <code>DELETE</code> statement without a <code>WHERE</code> clause will delete all rows in a table.</p>
</div>
<div class="paragraph">
<p>If that is truly what is intended, the <code>TRUNCATE</code> statement should be used because it is faster and uses fewer system and transaction log resources. Moreover when using the <code>TRUNCATE</code> statement, there is no ambiguity about the purpose of the SQL statement.</p>
</div>
<div class="paragraph">
<p>If removing all content from the table is not what is intended, a <code>WHERE</code> clause should be added.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">DELETE FROM COUNTRIES</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">TRUNCATE TABLE COUNTRIES</code></pre>
</div>
</div>
<div class="paragraph">
<p>or</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-text" data-lang="text">DELETE FROM COUNTRIES WHERE CODE = country_code</code></pre>
</div>
</div>
</div>
</div>