<div class="sect1">
<h2 id="_description">Description</h2>
<div class="sectionbody">
<div class="paragraph">
<p>Having several levels of nested SQL SELECT statements makes the code difficult to read and should therefore be avoided.</p>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_noncompliant_code_example">Noncompliant Code Example</h2>
<div class="sectionbody">
<div class="paragraph">
<p>With an allowed nesting level of 2:</p>
</div>
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">*&gt; Non-Compliant
EXEC SQL
  SELECT * FROM my_table1 WHERE
    my_column1 IN
      (SELECT my_column2 FROM my_table2
        WHERE my_column3 IN
          (SELECT my_column4 FROM my_table3))
END-EXEC.</code></pre>
</div>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_compliant_solution">Compliant Solution</h2>
<div class="sectionbody">
<div class="listingblock">
<div class="content">
<pre class="highlight"><code class="language-cobol" data-lang="cobol">EXEC SQL
  SELECT * FROM my_table
END-EXEC.</code></pre>
</div>
</div>
<hr>
</div>
</div>
<div class="sect1">
<h2 id="_implementation_specification">Implementation Specification</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_parameters">Parameters</h3>
<div class="sidebarblock">
<div class="content">
<div class="title">maximumNestedLevelThreshold</div>
<div class="listingblock">
<div class="content">
<pre>3</pre>
</div>
</div>
<div class="paragraph">
<p>The number of authorized nested "SELECT" statements</p>
</div>
</div>
</div>
<hr>
</div>
</div>
</div>
<div class="sect1">
<h2 id="_comments_and_links">Comments And Links</h2>
<div class="sectionbody">
<div class="paragraph">
<p>(visible only on this page)</p>
</div>
<div class="sect2">
<h3 id="_is_related_to_s2524">is related to: <a data-rspec-id="S2524" class="rspec-auto-link">S2524</a></h3>

</div>
</div>
</div>